### ISSUE TITLE
CONTENT001: False positives on 'bug' in normal prose

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#338]
## Description

CONTENT001 (no-todo-comments) is flagging the word "bug" when it appears in normal prose, not as a TODO-style comment.

## Examples

False positives from The Rust Book:
```
warning[CONTENT001]: BUG comment found - resolve before publishing
  --> ch02-00-guessing-game-tutorial.md:459:23
     |
 459 | contains an important bug fix, but it also contains a regression that will
```

```
warning[CONTENT001]: BUG comment found - resolve before publishing
  --> ch03-01-variables-and-mutability.md:43:17
     |
  43 | of this kind of bug can be difficult to track down after the fact, especially
```

These are discussing bugs conceptually, not marking TODO items.

## Expected Behavior

CONTENT001 should only match comment-style patterns:
- `// BUG:` or `// BUG `
- `/* BUG */`
- `<!-- BUG -->`
- `# BUG:` (in markdown)
- Maybe standalone `BUG:` at start of line

Should NOT match:
- "this bug" / "a bug" / "bug fix" in prose
- "debugging" / "debug"

## Found While Testing

Ran mdbook-lint against rust-lang/book - found 31 false positives for this rule.
